pkg/daemon/ceph/client/pool.go

```diff
@@ -510,18 +510,23 @@ func updatePoolCrushRule(context *clusterd.Context, clusterInfo *ClusterInfo, cl
 		return nil
 	}
 
-	if currentFailureDomain != pool.FailureDomain {
-		logger.Infof("creating a new crush rule for changed failure domain on crush rule %q", details.CrushRule)
-	}
-	if currentDeviceClass != pool.DeviceClass {
-		logger.Infof("creating a new crush rule for changed deviceClass on crush rule %q", details.CrushRule)
-	}
-
 	// Use a crush rule name that is unique to the desired failure domain
 	crushRuleName := fmt.Sprintf("%s_%s", pool.Name, pool.FailureDomain)
 	if pool.DeviceClass != "" {
 		crushRuleName = fmt.Sprintf("%s_%s_%s", pool.Name, pool.FailureDomain, pool.DeviceClass)
 	}
+	if crushRuleName == details.CrushRule {
+		logger.Debugf("crush rule already set to %q for pool %q", crushRuleName, pool.Name)
```

Comment:
We could also log failure domain and device class here informationally. It's definitely not necessary but could clarify Rook's internal state if there is ever a time when it doesn't fully match Ceph's state or the user's desired state.